### PR TITLE
Observe ViewEvents only when resumed

### LINKED ISSUE #7722: Replace calls to launchWhenResumed with repeatOnLifecycle
`launchWhenResumed` is not recommended and will be removed in the future.
We should instead use `repeatOnLifecycle`

Exemple: 
```
 lifecycleScope.launch {
                lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
                    flow.collect { flowItem ->
                        //DoSomethingWithFlowItem(flowItem
                    }
                }
            }
```

We might also be safer with listening `viewEvents` flow from ViewModel, to avoid refreshing UI if the app is in background: 
So replace:
```
viewEvents
                .stream()
                .onEach {
                    hideWaitingView()
                    observer(it)
                }
                .launchIn(lifecycleScope)
```

By:
```
lifecycleScope.launch { 
            repeatOnLifecycle(Lifecycle.State.RESUMED){
                viewEvents.stream()
                        .collect {
                            hideWaitingView()
                            observer(it)
                        }
            }
        }
```

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Observe ViewEvents only when the lifecycle is in the RESUMED state.

## Motivation and context

Closes #7722. `launchWhenResumed` is never used when observing flow, so there is no change to do on that.

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/MainActivity.kt [hunk 1]
```diff
@@ -142,9 +140,9 @@ class MainActivity : VectorBaseActivity<ActivityMainBinding>(), UnlockedActivity
         startAppViewModel.onEach {
             renderState(it)
         }
-        startAppViewModel.viewEvents.stream()
-                .onEach(::handleViewEvents)
-                .launchIn(lifecycleScope)
+        startAppViewModel.observeViewEvents(Lifecycle.State.CREATED) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/MainActivity.kt:143]: Need to use CREATED here, or the app does not start (race condition?)

Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/MainActivity.kt:143]: It's really weird no? What exactly do you mean by doesn't start?

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/MainActivity.kt:143]: The app stays in the MainActivity forever, the event `AppStarted` is sent but not received.

**Review by ganfra — COMMENTED** [review]
Where are we with these changes?

**Comment by bmarty** [comment]
> Where are we with these changes?

PR updated, happy to merge now that the release is out.

**Comment by bmarty** [comment]
(will take care of the test compilation issue)

**Review by ganfra — APPROVED** [review]
LGTM

**Pull request merged**